datasets/ethos/ethos.py

```diff
@@ -0,0 +1,159 @@
+detection on social media platforms, called Ethos. There are two variations of the dataset:
+
+Ethos_Dataset_Binary: contains 998 comments in the dataset alongside with a label
+about hate speech presence or absence. 565 of them do not contain hate speech,
+while the rest of them, 433, contain.
+
+Ethos_Dataset_Multi_Label: which contains 8 labels for the 433 comments with hate speech content.
+These labels are violence (if it incites (1) or not (0) violence), directed_vs_general (if it is
+directed to a person (1) or a group (0)), and 6 labels about the category of hate speech like,
+gender, race, national_origin, disability, religion and sexual_orientation.
+"""
+
+_URL = "https://github.com/intelligence-csd-auth-gr/Ethos-Hate-Speech-Dataset"
+
+
+class EthosConfig(datasets.BuilderConfig):
+    """BuilderConfig for Ethos."""
+
+    def __init__(self, variation="binary", **kwargs):
+        """Constructs an EthosDataset.
+
+        Args:
+        variation: can be binary or multilabel
+        **kwargs: keyword arguments forwarded to super.
+        """
+        if variation.lower() == "binary":
+            self.variation = "binary"
+        elif variation.lower() == "multilabel":
+            self.variation = "multilabel"
+        else:
+            logging.warning("Wrong variation. Could be either 'binary' or 'multilabel', using 'binary' instead.")
+            self.variation = "binary"
+        super(EthosConfig, self).__init__(**kwargs)
+
+
+class Ethos(datasets.GeneratorBasedBuilder):
+    BUILDER_CONFIG_CLASS = EthosConfig
+
+    BUILDER_CONFIGS = [
+        EthosConfig(
+            name="binary",
+            version=datasets.Version("1.0.0", ""),
+            description="Ethos Binary",
+            variation="binary",
+        ),
+        EthosConfig(
+            name="multilabel",
+            version=datasets.Version("1.0.0", ""),
+            description="Ethos Multi Label",
+            variation="multilabel",
+        ),
+    ]
+
+    def _info(self):
+        if self.config.variation == "binary":
+            f = datasets.Features(
+                {
+                    "text": datasets.Value("string"),
+                    "label": datasets.features.ClassLabel(names=["no_hate_speech", "hate_speech"]),
+                }
+            )
+        else:
+            f = datasets.Features(
+                {
+                    "text": datasets.Value("string"),
+                    "violence": datasets.Value("int32"),
+                    "directed_vs_generalized": datasets.Value("int32"),
+                    "gender": datasets.Value("int32"),
+                    "race": datasets.Value("int32"),
+                    "national_origin": datasets.Value("int32"),
+                    "disability": datasets.Value("int32"),
+                    "religion": datasets.Value("int32"),
+                    "sexual_orientation": datasets.Value("int32"),
```

Comment:
This fields are true or false right ?

Then can you use the ClassLabel feature type here for those fields ?
For example
```
"race": datasets.ClassLabel(names=["false", "true"])
```

You'll need to regenerate the dataset_infos.json file after this change

Reply:
Done!